scripts/missions/asa/04_Sugar-Coated_Directive.lua

```diff
@@ -0,0 +1,277 @@
+                        mission:setVar(player, 'Leviathan', 1)
+                        return mission:messageSpecial(tidesID.text.POWER_STYMIES, xi.keyItem.DOMINAS_CERULEAN_SEAL)
+                    elseif player:hasKeyItem(xi.ki.DOMINAS_CERULEAN_SEAL) and asaPrimal == 2 then
+                        mission:setVar(player, 'Leviathan', 3)
+                        return mission:progressEvent(2)
+                    end
+                end,
+
+                onEventFinish =
+                {
+                    [2] = function(player, csid, option, npc)
+                        player:delKeyItem(xi.ki.DOMINAS_CERULEAN_SEAL)
+                        player:addKeyItem(xi.ki.CERULEAN_COUNTERSEAL)
+                        player:messageSpecial(tidesID.text.ATTACH_SEAL, xi.ki.DOMINAS_CERULEAN_SEAL)
+                        player:messageSpecial(tidesID.text.KEYITEM_OBTAINED, xi.ki.CERULEAN_COUNTERSEAL)
+                        local sealCount = mission:getVar(player, 'SealCount')
+                        mission:setVar(player, 'SealCount', sealCount + 1)
+                    end,
+                }
+            },
+        },
+
+        [xi.zone.CLOISTER_OF_TREMORS] =
+        {
+            ['Earth_Protocrystal'] =
+            {
+                onTrigger = function(player, npc)
+                    local asaPrimal = mission:getVar(player, 'Titan')
+
+                    if player:hasKeyItem(xi.ki.DOMINAS_AMBER_SEAL) and asaPrimal == 0 then
+                        mission:setVar(player, 'Titan', 1)
+                        return mission:messageSpecial(tremorsID.text.POWER_STYMIES, xi.keyItem.DOMINAS_AMBER_SEAL)
+                    elseif player:hasKeyItem(xi.ki.DOMINAS_AMBER_SEAL) and asaPrimal == 2 then
+                        mission:setVar(player, 'Titan', 3)
+                        return mission:progressEvent(2)
+                    end
+                end,
+
+                onEventFinish =
+                {
+                    [2] = function(player, csid, option, npc)
+                        player:delKeyItem(xi.ki.DOMINAS_AMBER_SEAL)
+                        player:addKeyItem(xi.ki.AMBER_COUNTERSEAL)
+                        player:messageSpecial(tremorsID.text.ATTACH_SEAL, xi.ki.DOMINAS_AMBER_SEAL)
+                        player:messageSpecial(tremorsID.text.KEYITEM_OBTAINED, xi.ki.AMBER_COUNTERSEAL)
+                        local sealCount = mission:getVar(player, 'SealCount')
+                        mission:setVar(player, 'SealCount', sealCount + 1)
+                    end,
+                }
+            },
+        },
+
+        [xi.zone.QUFIM_ISLAND] =
+        {
+            ['Trodden_Snow'] =
+            {
+                onTrigger = function(player, npc)
+                    --ASA 4 CS: Triggers With At Least 3 Counterseals.
+                    local completedSeals =
+                        (player:hasKeyItem(xi.ki.AMBER_COUNTERSEAL)    and 1 or 0) +
```

Comment:
Since you're just counting KIs here, the KIs should be in a table, and loop over them to count.